tests/sles4sap/robot_fw.pm

```diff
@@ -20,6 +20,29 @@ use version_utils qw(is_sle);
 use utils qw(zypper_call);
 use hacluster qw(is_package_installed);
 
+sub remove_value {
+    my ($module, $parameter) = @_;
+    assert_script_run "perl -e 'while (<>) { /testcase.+name=\"([^\"]+)\"/; \$testcase = \$1; unless (\$testcase eq $parameter) { print; } }' < $module.xml > result.xml";
+    # As we just removed a failed value, we have to decrease the failure counter by 1.
+    assert_script_run 'eval $(awk \'/failures=/ {print $5}\' result.xml); new=$((failures-1)); sed -i "s/failures=\"$failures\"/failures=\"$new\"/" result.xml';
```

Comment:
If the `awk` command fails, full shell command can still return 0 and give a successful result for `assert_script_run` even though `result.xml` file content is not changed. However I have not found a better way to do this. Will update my comment if/when I do.

Reply:
You can do `&&` instead `;` after awk

Reply:
Tried with `&&` but it can still return 0. The issue is that `awk` is run in between `$( )` so that catches the return value, and then the `eval` block returns 0 is its input is empty, which it would be if the `awk` command fails. I tried wrapping it in a `bash -e`, but it does not work either.

Reply:
Why you need to use `eval` or `$()` ?

Reply:
`awk \'/failures=/ {print $5}\' result.xml && new=$((failures-1)); sed -i "s/failures=\"$failures\"/failures=\"$new\"/" result.xml`

Reply:
> Why you need to use `eval` or `$()` ?

The `awk` code in between `$( )` is returning a text formatted as `failures="3"`. When this is passed to `eval`, it creates a variable named `failures` that's accessible from the shell, and which is being used by `new=$((failures-1))`.

> `awk \'/failures=/ {print $5}\' result.xml && new=$((failures-1)); sed -i "s/failures=\"$failures\"/failures=\"$new\"/" result.xml`
> Should work or ?

Without the `eval` you get an error like this:

```
# $(awk '/failures=/ {print $5}' result.xml) && new=$((failures-1)) && sed -i 's/failures=\"$failures\"/failures=\"$new\"/' result.xml
If 'failures="2"' is not a typo you can use command-not-found to lookup the package that contains it, like this:
    cnf failures="2"
# echo $?
127
```

And without the `eval` and the `$( )` you get an output like this:

```
# awk '/failures=/ {print $5}' result.xml2 && new=$((failures-1)) && sed -i 's/failures=\"$failures\"/failures=\"$new\"/' result.xml
awk: fatal: cannot open file `result.xml2' for reading (No such file or directory)
# echo $?
2
# awk '/failures=/ {print $5}' result.xml && new=$((failures-1)) && sed -i 's/failures=\"$failures\"/failures=\"$new\"/' result.xml
failures="3"
# echo $?
0
```
Command works in the sense that it will capture the correct return value of the shell command line if `awk` fails, but it does not modify the contents of the file:

```
# diff robot_fw-sysctl.robot.xml result.xml ; grep failures robot_fw-sysctl.robot.xml result.xml 
189,191d188
< <testcase classname="Sysctl" name="Sysctl_kernel_panic_on_oops" time="0.008">
< <failure message="expected: 0&#10;returned: 1&#10;invalid value: '1', expected '0'" type="AssertionError"></failure>
< </testcase>
robot_fw-sysctl.robot.xml:<testsuite name="Sysctl" tests="818" errors="0" failures="3" skipped="0" time="7.713">
result.xml:<testsuite name="Sysctl" tests="818" errors="0" failures="3" skipped="0" time="7.713">
```

My suggestion would be to use the `inplace` module of gawk as this:

```
awk -i inplace '/failures=/ { new=substr($5,11,length($5)-11); new--; gsub($5, "failures=\""new"\"") } /./ { print }' result.xml
```

This works in Leap. I did not test in SLES.

Reply:
This way everything is in a single `awk` command. If it fails, `assert_script_run` will abort the test.

Reply:
There must be failure, this will be executed only when `check_failure` passed